crates/uv/tests/it/lock.rs

```diff
@@ -2164,6 +2164,1147 @@ fn lock_dependency_non_existent_extra() -> Result<()> {
     Ok(())
 }
 
+/// This tests a "basic" case for specifying conflicting extras.
+///
+/// Namely, we check that 1) without declaring them conflicting,
+/// resolution fails, 2) declaring them conflicting, resolution
+/// succeeds, 3) install succeeds, 4) install fails when requesting two
+/// or more extras that are declared to conflict with each other.
+///
+/// This test was inspired by:
+/// <https://github.com/astral-sh/uv/issues/8024>
+#[test]
+fn lock_conflicting_extra_basic() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    // First we test that resolving with two extras that have
+    // conflicting dependencies fails.
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        description = "Add your description here"
+        requires-python = ">=3.12"
+
+        [project.optional-dependencies]
+        project1 = ["numpy==1.26.3"]
```

Comment:
can we use something less heavy than numpy? numpy is a compiled wheel and therefore rather large.

Reply:
Done! I switched over to `sortedcontainers`.